### PR TITLE
Test if ignore prevails over restrict

### PR DESCRIPTION
**Pull request by jesusjimsa**
| Related issue |
|---|
| #326 |

# Description
Added test to check if the `ignore` tag prevails over the `restrict` one when using both in the same directory.  For example:

```XML
<directories restrict="file_name">/home</directories>
<ignore>file_name</ignore>
```
With this configuration syscheck must ignore the files called `file_name`.

This test covers strings and sregex cases.

I have also modified `fim.py` to support both types of configurations: ignore by string and ignore by sregex.

## Platforms supported by the test
- [x] Linux
- [x] macOS
- [x] Windows

### CODE REVIEW

**Comment by jesusjimsa** [comment]
I have also modified `fim.py` to support both types of configurations: ignore by string and ignore by sregex.

**Comment by jesusjimsa** [comment]
# Results
## Linux
```
===================================================================== test session starts ======================================================================
platform linux -- Python 3.6.8, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
rootdir: /home/vagrant/qa/test_wazuh, inifile: pytest.ini
collected 30 items

test_ignore/test_ignore_works_over_restrict.py ..sssss.....sssss.....sssss...                                                                            [100%]

==================================================== 15 passed, 15 skipped in 234159.39s (2 days, 17:02:39) ====================================================
```
## Windows
```
================================================= test session starts ================================================== 
platform win32 -- Python 3.7.3, pytest-5.1.2, py-1.8.0, pluggy-0.13.0
rootdir: C:\Users\jmv74211\Desktop\wazuh-qa\test_wazuh, inifile: pytest.ini
collected 30 items
test_fim\test_ignore\test_ignore_works_over_restrict.py ..sssss.....sssss.....sssss...                            [100%]
================================ 15 passed, 15 skipped in 234148.75s (2 days, 17:02:28) ================================
```
## macOS
```
==================================================== test session starts =====================================================
platform darwin -- Python 3.7.5, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
rootdir: /Users/vagrant/qa/test_wazuh, inifile: pytest.ini
collected 10 items

test_ignore/test_ignore_works_over_restrict.py ..sssss...                                                              [100%]

==================================== 5 passed, 5 skipped in 233964.59s (2 days, 16:59:24) ====================================
```

**Comment by jesusjimsa** [comment]
# Test ignore_valid
```
==================== 354 passed, 336 skipped in 5523995.30s (63 days, 22:26:35) ====================
```

File: test_wazuh/test_fim/test_ignore/test_ignore_works_over_restrict.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import pytest
+
+from wazuh_testing.fim import LOG_FILE_PATH, callback_ignore, callback_detect_event, create_file, REGULAR, \
+    generate_params, check_time_travel, DEFAULT_TIMEOUT
+from wazuh_testing.tools import FileMonitor, check_apply_test, load_wazuh_configurations, PREFIX
+
+# Markers
+
+pytestmark = [pytest.mark.linux, pytest.mark.darwin, pytest.mark.win32]
+
```
Comment (crd1985) [inline test_wazuh/test_fim/test_ignore/test_ignore_works_over_restrict.py]: Delete these lines

File: test_wazuh/test_fim/test_ignore/test_ignore_works_over_restrict.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+@pytest.mark.parametrize('folder, filename, triggers_event, tags_to_apply', [
+    (testdir1, 'testfile', False, {'valid_no_regex'}),
+    (testdir2, 'not_ignored_string', True, {'valid_no_regex'}),
+    (testdir1, 'testfile2', False, {'valid_regex'}),
+    (testdir1, 'ignore_testfile2', False, {'valid_regex'}),
+    (testdir2, 'not_ignored_sregex', True, {'valid_regex'})
+])
+def test_ignore_works_over_restrict(folder, filename, triggers_event, tags_to_apply, get_configuration,
+                                    configure_environment, restart_syscheckd, wait_for_initial_scan):
+    """ Checks if the ignore tag prevails over the restrict one when using both in the same directory.
```
Comment (crd1985) [inline test_wazuh/test_fim/test_ignore/test_ignore_works_over_restrict.py]:
```suggestion
    """Checks if the ignore tag prevails over the restrict one when using both in the same directory.
```

**Pull request merged**